packages/trader/src/sass/app/_common/components/purchase-button.scss

```diff
@@ -84,7 +84,7 @@
         }
     }
     &__text {
-        @include typeface(--paragraph-left-bold-active);
+        color: var(--general-active);
```

Comment:
Why do we need to keep `color` in some cases?

Reply:
because we are only using colors in theme.scss which are related to text and start with `--text`. some codes are not following this so I keep the className related to it.

Reply:
Is this good? As I know the purpose of adding a text-specific component is to encapsulate text styles in the component and remove the external CSS. Right?
How about sending full color names instead of just text-specific ones?

Reply:
I do agree. Let me check with other teammates.

Reply:
Of course, the problem is that only those colors are for text and I donno why are we using other colors.

Reply:
About the point you mentioned above, yes you are right, we wanna reduce css files, but sometimes we have to keep the style because at that time we didn't have this component and the codes have been written in anyway. So from now on, we can keep the regulation. 
Anyway, I will check this problem to see if we can solve it. Thank you.

Reply:
I think this relates more to design than us. If they want to use a new color for texts, they should update color palette too but I'm not aware if they did or not.

Reply:
i dont think design ever use general-active for text color. its most likely a mistake from us to put general active here, can check in abstract for this. may i know which section is this?

Reply:
I agree with @salar-fs and I think we should keep one paradigm everywhere and avoid special cases as much as possible.
But if we have no choice to handle special cases, we can keep the beauty of  both worlds in this way in the `Text` component (Actually, I am against this myself! this is just a last solution if other solutions are impossible):

```js
let color = colorProp.startsWith('--') ? `var(${colorProp})` : `var(--text-${colorProp})`
```

Reply:
I guess passing the color down to the component in case of unusual colors like this (which has to be always white) is not a bad practice.

Reply:
Yes @reza-fs , I considered all unsuall properties - which is out of our regulations - as a className.